dashboard/test/models/user_test.rb

```diff
@@ -5473,6 +5473,70 @@ def hide_scripts_in_sections(section1, section2)
     assert_equal new_us_state, student.reload.us_state
   end
 
+  describe 'strict password requirements by country' do
+    let(:strict_country) {User::PASSWORD_STRICT_COUNTRIES.first}
+
+    before do
+      allow(DCDO).to receive(:get).with('strict-password-country', false).and_return(true)
+      allow(DCDO).to receive(:get).with('migration_service_enabled', false).and_return(false)
+    end
+
+    describe 'when creating a teacher in a strict country' do
+      context 'when using a short password' do
+        let(:user) {build(:teacher, country_code: strict_country, password: 'short', password_confirmation: 'short')}
+        it 'rejects passwords' do
+          _(user).wont_be :valid?
+        end
+
+        it 'adds an error message' do
+          user.valid?
+          _(user.errors[:password]).must_include 'is too short (minimum is 14 characters)'
+        end
+      end
+      context 'when using a short password' do
+        let(:user) {create(:teacher, country_code: strict_country, password: 'longlongpassword', password_confirmation: 'longlongpassword')}
+
+        it 'accepts passwords that are at least 14 characters' do
+          _(user).must_be :valid?
+        end
+
+        it 'has no password errors for long password' do
+          user.valid?
+          _(user.errors[:password]).must_be :empty?
+        end
+      end
+    end
+
+    describe 'when updating a teacher in a strict country' do
+      let(:teacher) {build(:teacher, country_code: strict_country)}
+
+      it 'rejects a too-short password' do
+        result = teacher.update(password: 'tooshort', password_confirmation: 'tooshort')
+        _(result).must_equal false
+      end
+
+      it 'adds error for too-short password' do
+        teacher.update(password: 'tooshort', password_confirmation: 'tooshort')
+        _(teacher.errors[:password]).must_include 'is too short (minimum is 14 characters)'
+      end
+
+      it 'accepts a sufficiently long password' do
+        result = teacher.update(password: 'longlongpassword', password_confirmation: 'longlongpassword')
+        _(result).must_equal true
+      end
+
+      it 'has no password errors after successful update' do
+        teacher.update(password: 'longlongpassword', password_confirmation: 'longlongpassword')
+        _(teacher.errors[:password]).must_be :empty?
+      end
+
+      it 'persists the new password' do
+        teacher.update(password: 'longlongpassword', password_confirmation: 'longlongpassword')
+        _(teacher.password).must_equal 'longlongpassword'
```

Comment:
I am reviewing all these test, and I know I have told you before to try keeping one assertion per test, but I was reading over the recommendations [here](https://www.betterspecs.org/#single) and in this case, we may benefit from grouping assertions for performance reasons.

> in tests that are not isolated (e.g. ones that integrate with a DB, an external webservice, or end-to-end-tests), you take a massive performance hit to do the same setup over and over again, just to set a different expectation in each test. In these sorts of slower tests, I think it's fine to specify more than one isolated behavior.